### ISSUE TITLE
[FR] nested form support

### ISSUE THREAD

**Issue by ssingar** [ampproject/amphtml#15878]
Hi,
   I am trying to work on a promotional email using Google AMP and I need some inputs which I am missing. Please help me here.

 I am trying to send a promotional email once user provides email Id. This is a two step process.
1. Make a call to ValidateEmail to validate the email if its present in DB or not.
2. Make second call SignUpForSpecialOffers only when first call is successful by passing the email id. The logic for the  second call is inside the submit-sucess, but still I am not able to make the second call. Please let me know if I am missing anything here or if there are better approach?

<form method="POST" action-xhr="/ValidateEmail" target="_top" custom-validation-reporting="show-first-on-submit" class="clearfix">
	<div class="email-signup-form">
	<input type="email" class="ampstart-input" id="submit-email" name="emailAddress" 
	pattern="[a-zA-Z0-9_]+(?:\.[A-Za-z0-9!#$%&amp;&#x27;*+/=?^_`{|}~-]+)*@(?!([a-zA-Z0-9]*\.[a-zA-Z0-9]*\.[a-zA-Z0-9]*\.))(?:[A-Za-z0-9](?:[a-zA-Z0-9-]*[A-Za-z0-9])?\.)+[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?" 
	placeholder="Enter your email address" required=""/>
	<input type="submit" value="Submit" class="ampstart-btn"/>
	</div>
	<span visible-when-invalid="typeMismatch" validation-for="submit-email" class="error-block">
		<div class='form-error margintop'>The email address isn't valid. Please check the format and try again.</div>
	</span>
	<div submit-error>
		<template type="amp-mustache">
			{{^exception}}
                <div class='form-error margintop'>{{errorMessage}}</div>
            {{/exception}}
         </template>
 	</div>
 	<div submit-success>
 		<template type="amp-mustache">
			<c:set var="emailId">{{emailAddress}}</c:set>         	
         	</template>
 		<c:set var="prefCenterURL" value="https://mysite.com/pub/rf?_ri_=X0G&EMAIL_PERMISSION_STATUS=I&EMAIL_ADDRESS_=${emailId}"/>
 		<form method="POST" id="splOfferEmailId" action-xhr="/SignUpForSpecialOffers" target="_top" class="clearfix">
			<input type="hidden" name="email" value="${emailId}"/>
			<input type="text" name="splOffer" class="ampstart-btn" on="change:splOfferEmailId.submit"/>
			<div submit-error>
				<template type="amp-mustache">
					{{^exception}}
		                <div class='form-error margintop'>{{errorMessage}}</div>
		            {{/exception}}	         	
		         </template>
		 	</div>
		 	<div submit-success>
		 		<template type="amp-mustache">
			 		{{^status}}
			 		<div class="after-signup">
			 			<div class="thanks">Thanks for signing up.</div>
			 		</div>
			 		{{/status}}
		 		</template>
		 	</div>
 		</form>
 	</div>
</form>

**Comment by jaygray0919** [comment]
Can't help without seeing the amp-list (JSON) being parsed by amp-mustache

**Comment by ssingar** [comment]
@jaygray0919 ,
Here is the JSON response for the first call:

{
	"status": "valid",
	"catalogId": ["10001"],
	"emailAddress": ["test123@abc.com"],
	"suggestedEmail": "",
	"langId": ["-1"],
	"storeId": ["10101"],
	"code": ""
}

Also, the expected response for second call will be:
{"success":"true"}

**Comment by aghassemi** [comment]
@ssingar I recommend using the [AMP Slack channel](https://www.ampproject.org/support/developer/) or Stackoverflow for questions and reserving GitHub for bug reports.

Thanks.

**Comment by ericlindley-g** [comment]
It seems like the feature request here is to make additional server calls to different endpoints via amp-form, conditional on the success/failure of previous calls.

@ssingar , is there a reason you don't set up this logic on the server side? Naively I would expect the client to just submit one request, and that the server would handle the rest, but I know that some folks have dependencies in their stack that require workarounds, even if it's not strictly a best practice.

**Comment by cvializ** [comment]
The HTML spec does not allow nested forms, so I don't think AMP should support them.

https://html.spec.whatwg.org/multipage/forms.html#the-form-element

> Content model:
> Flow content, but with no form element descendants.

**Comment by cvializ** [comment]
You can use `verify-xhr` to check for a valid email address before submitting https://amp.dev/documentation/components/amp-form#verification
